extensions/objects/templates/svg-filename-editor.component.ts

```diff
@@ -598,6 +600,97 @@ angular.module('oppia').component('svgFilenameEditor', {
         createPolygon();
       };
 
+      var makeCurveCircle = function(left, top) {
+        var size = ctrl.fabricjsOptions.size;
+        var circle = new fabric.Circle({
+          left: left,
+          top: top,
+          radius: parseInt(size.substring(0, size.length - 2)) + 2,
+          fill: '#666666',
+          stroke: '#666666',
+          hasBorders: false,
+          hasControls: false
+        });
+        return circle;
+      };
+
+      var drawQuadratic = function() {
+        var size = ctrl.fabricjsOptions.size;
+        var curve = new fabric.Path('M 40 40 Q 95, 100, 150, 40', {
+          stroke: ctrl.fabricjsOptions.stroke,
+          fill: ctrl.fabricjsOptions.fill,
+          strokeWidth: parseInt(size.substring(0, size.length - 2)),
+          objectCaching: false,
+          selectable: false
+        });
+        ctrl.canvas.add(curve);
+
+        var p1 = makeCurveCircle(95, 100);
+        p1.name = 'p1';
+        p1.set({
+          radius: 12,
+          fill: '#ffffff',
+          strokeWidth: 5
+        });
+        ctrl.canvas.add(p1);
+
+        var p0 = makeCurveCircle(40, 40);
+        p0.name = 'p0';
+        ctrl.canvas.add(p0);
+
+        var p2 = makeCurveCircle(150, 40);
+        p2.name = 'p2';
+        ctrl.canvas.add(p2);
+      };
+
+      ctrl.createBezier = function() {
+        if (ctrl.drawMode === DRAW_MODE_NONE) {
+          ctrl.canvas.discardActiveObject();
+          ctrl.drawMode = DRAW_MODE_BEZIER;
+          ctrl.canvas.getObjects().forEach(function(item) {
+            item.set({
+              hoverCursor: 'default',
+              selectable: false
+            });
+          });
+          drawQuadratic();
+        } else {
+          // Remove the circles and the current path.
```

Comment:
Is this the case where the user clicks on the bezier tool again after drawing? Please add comments.

Reply:
Yes. Done.